### PR TITLE
Make it easier to batch evaluate pure MCTS runs

### PR DESCRIPTION
**Pull request by nikihowe**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## TODO

- make the evaluation against each MCTS difficulty run in a separate docker container (two advantages: doesn't rely on code in current devbox, and also can run in parallel)

### CODE REVIEW

File: experiments/neural_mcts/eval_scripts/plotting/make_plots.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import matplotlib.pyplot as plt
```
Comment (rhaps0dy) [inline experiments/neural_mcts/eval_scripts/plotting/make_plots.py:1]: Why not run `make_the_plot` from `old_make_plots.py` ?

Comment (rhaps0dy) [reply experiments/neural_mcts/eval_scripts/plotting/make_plots.py:1]: (just consider it, no need to write out a justification)

File: experiments/neural_mcts/eval_scripts/plotting/utils.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import re
+
+
+def get_model_and_epoch(model_line):
```
Comment (rhaps0dy) [inline experiments/neural_mcts/eval_scripts/plotting/utils.py:4]: Are you parsing the command line to get the model and epoch? Or is this parsed from the log string?

Why not just grab these values from the arguments json or something? Parsing is brittle

File: experiments/pure_mcts/run_experiments_from_files.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import os
+import shlex
+import subprocess
+import sys
+
+from run_given_experiment import run_from_command_line_inputs
+
+
+def get_docker_commands_from_directories(root_directory):
```
Comment (rhaps0dy) [inline experiments/pure_mcts/run_experiments_from_files.py]: What's the difference between this and the other PR?

File: experiments/neural_mcts/eval_scripts/plotting/make_plots.py (hunk 1 above)
Comment (nikihowe) [reply experiments/neural_mcts/eval_scripts/plotting/make_plots.py:1]: at some point we will want to put them together. We need the functionality of both as we transition to better checkpoint naming (which is set up for new runs).

File: experiments/pure_mcts/run_experiments_from_files.py (hunk 3 above)
Comment (nikihowe) [reply experiments/pure_mcts/run_experiments_from_files.py]: this one has new code which lets us automate the pure MCTS runs

File: experiments/neural_mcts/eval_scripts/plotting/utils.py (hunk 2 above)
Comment (nikihowe) [reply experiments/neural_mcts/eval_scripts/plotting/utils.py:4]: yes I am parsing the CLI output. I bet there could be cleaner solutions. Do you have any suggestions?

Comment (nikihowe) [reply experiments/neural_mcts/eval_scripts/plotting/utils.py:4]: tbh I haven't looked at this code in a little while and I'm not sure why it's apparently not used.

**Review by rhaps0dy — APPROVED** [review]
LGTM

**Pull request merged**